### ISSUE TITLE
No "Enter/Exit Live Reveal Slideshow" button

### ISSUE THREAD

**Issue by Vaits** [damianavila/RISE#275]
Hello, 
I can select "Slideshow" via "Cell Toolbar" but there is no "Enter/Exit Live Reveal Slideshow" button for me. I installed RISE using `conda install -c damianavila82 rise` (Python 3.6). To further investigate, I hard refreshed my browser, used different browsers (firefox and chromium) and created a new python 3.5 environment. Nothing changed, there is still no button and I'm not able to start my presentations. Appreciating any help.

`conda create -n rise35 python=3.5 anaconda`
`conda install -n rise35 -c damianavila82 rise`
`source activate rise35`

`jupyter nbextension list`

```
Known nbextensions:
  config dir: /home/vaits/anaconda3/envs/rise35/etc/jupyter/nbconfig
    notebook section
      jupyter-js-widgets/extension  enabled 
      - Validating: OK
      rise/main  enabled 
      - Validating: OK
```

**Comment by Vaits** [comment]
I just had to create new jupyter notebooks. Now it is working fine in all environments. My old notebooks are still missing RISE. Is this intended? Maybe a small hint in the README would be useful.

**Comment by damianavila** [comment]
> I just had to create new jupyter notebooks. Now it is working fine in all environments

That weird...

> My old notebooks are still missing RISE. Is this intended?

Nop, do you have any error in th js console when you are dealing with this old notebooks?

**Comment by Vaits** [comment]
Yes, I'm only getting these errors using the old notebooks.

```
Uncaught TypeError: define is not a function
    at :8888/nbextensions/nb_conda/main.js?v=20170720142008:1

Uncaught TypeError: define is not a function
    at :8888/nbextensions/rise/main.js?v=20170720142008:11

Uncaught TypeError: define is not a function
    at extension.js:1

Access to Font at 'about:blank' from origin 'http://localhost:8888' has been blocked by CORS policy: Invalid response. Origin 'http://localhost:8888' is therefore not allowed access.`

**Comment by damianavila** [comment]
Can you post one of those failing notebook? To see if I can reproduce the issue and investigate further...

**Comment by Vaits** [comment]
To create a simple example, I deleted all content from an old notebook. Then RISE was working properly. In the end I noticed, that there was only one code cell, which caused the errors. You can get the notebook <a href="https://www.dropbox.com/s/0887gbvmap6l82e/test5.ipynb?dl=1">Here</a>.

**Comment by mjbright** [comment]
I'm seeing a similar issue, let me know if you prefer me to open this as a separate issue.

I installed RISE under anaconda, with Python 3.6 using
```conda install -c damianavila82 rise```

but I didn't have the "Enter/Exit Slideshow" button in my notebook.
After seeing this post I created a new empty notebook which had the same problem.

In the notebook log output I saw errors.
I could create the same errors using
```jupyter nbextension list```:

```jupyter nbextension list            
Known nbextensions:                                                                                      
Traceback (most recent call last):                  
  File "/usr/bin/jupyter-nbextension", line 6, in <module>                                               
    main()                                                 
  File "/usr/lib/python3.6/site-packages/jupyter_core/application.py", line 267, in launch_instance      
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/usr/lib/python3.6/site-packages/traitlets/config/application.py", line 658, in launch_instance  
    app.start()                                     
  File "/usr/lib/python3.6/site-packages/notebook/nbextensions.py", line 988, in start                   
    super(NBExtensionApp, self).start()                           
  File "/usr/lib/python3.6/site-packages/jupyter_core/application.py", line 256, in start                
    self.subapp.start()                                                                                  
  File "/usr/lib/python3.6/site-packages/notebook/nbextensions.py", line 960, in start                   
    self.list_nbextensions()                            
  File "/usr/lib/python3.6/site-packages/notebook/nbextensions.py", line 943, in list_nbextensions       
    data = cm.get(section)                                                                               
  File "/usr/lib/python3.6/site-packages/traitlets/config/manager.py", line 63, in get                      
    return json.load(f)                             
  File "/usr/lib64/python3.6/json/__init__.py", line 299, in load                                                                                                                                                  
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads                                              
    return _default_decoder.decode(s)                                                                    
  File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode                                                                        
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)               
json.decoder.JSONDecodeError: Expecting ':' delimiter: line 14 column 15 (char 205)
```

I ran through my config json files using the following to find the culprit:
```for JSONF in $(find ~/.jupyter/ ~/.local/share/jupyter/ -iname '*.json' -type f); do echo $JSONF; python -mjson.tool  < $JSONF >/dev/null; done```

The culprit was ~/.jupyter/nbconfig/notebook.json which contained:
```
{                                                                                                                                                                                                                  
  "Cell": {                                                                                                                                                                                                        
    "cm_config": {                                                                                                                                                                                                 
      "lineNumbers": false                                                                                                                                                                                         
    }                                                                                                                                                                                                              
  },                                                                                                                                                                                                               
  "Notebook": {                                                                                                                                                                                                    
    "Toolbar": true,                                                                                                                                                                                               
    "Header": true                                                                                                                                                                                                 
  },                                                                                                                                                                                                               
  "load_extensions": {                                                                                                                                                                                             
    "nbbrowserpdf/index": true                                                                                                                                                                                     
  },                                                                                                                                                                                                               
  "livereveal", {                                                                                                                                                                                                  
      "theme": "sky",                                                                                                                                                                                              
      "transition": "zoom",                                                                                                                                                                                        
      "start_slideshow_at": "selected",                                                                                                                                                                            
  }                                                                                                                                                                                                                
}   
```

Changing this to 
```
{                                                                                                                                                                                                                  
  "Cell": {                                                                                                                                                                                                        
    "cm_config": {                                                                                                                                                                                                 
      "lineNumbers": false                                                                                                                                                                                         
    }                                                                                                                                                                                                              
  },                                                                                                                                                                                                               
  "Notebook": {                                                                                                                                                                                                    
    "Toolbar": true,                                                                                                                                                                                               
    "Header": true                                                                                                                                                                                                 
  },                                                                                                                                                                                                               
  "load_extensions": {                                                                                                                                                                                             
    "nbbrowserpdf/index": true                                                                                                                                                                                     
  },                                                                                                                                                                                                               
  "livereveal": {                                                                                                                                                                                                  
      "theme": "sky",                                                                                                                                                                                              
      "transition": "zoom",                                                                                                                                                                                        
      "start_slideshow_at": "selected"                                                                                                                                                                             
  }                                                                                                                                                                                                                
} 
```
fixed the problem

**Comment by damianavila** [comment]
Which platform, browser and notebook version are you using @mjbright, I would love to be able to reproduce this behavior.

**Comment by mjbright** [comment]
Hi, I'm running

- Anaconda  (conda 4.3.29),
- Python 3.6
- Chrome 63.0.3239.84 (Official Build) (64-bit) on
- Fedora 27
- notebook version 5.2.2 py36_1

Hope that helps, thanks for a great tool !

**Comment by damianavila** [comment]
@Aleczhang13 

Some questions:

1) notebook version?
2) browser version?
3) can you try hard-reloading your browser (some info here: https://www.getfilecloud.com/blog/2015/03/tech-tip-how-to-do-hard-refresh-in-browsers/#.Wnmm25M-dhE, to get rid of the cache)?
4) can you try installing in a new environment without other extensions to get rid of possible interferences?

Particularly, step 4 has been a successful fix for the issue for other people (#341).

**Comment by deepakagrawal** [comment]
I am facing same issue. I don't see the "Enter/Exit Live Reveal Slideshow" button.
I am using 

1. notebook v 5.0.0
2. browser Google Chrome Version 63.0.3239.132 (Official Build) (64-bit)

**Comment by deepakagrawal** [comment]
Looks like it was an issue with the notebook version. I updated notebook and jupyter using conda. After that I reinstalled rise using conda and then it worked fine. It even works for the older notebooks.

**Comment by damianavila** [comment]
Great, now I would like more info to be able to reproduce the issue, if possible.
Can you confirm the notebook version not working was 5.0.0? And it is working OK with 5.4.0?

**Comment by deepakagrawal** [comment]
Correct.

**Comment by damianavila** [comment]
OK, thanks for the additional information. Closing now.
